packages/chain-evmos/src/implementation.ts

```diff
@@ -0,0 +1,168 @@
+import {
+	Action,
+	ActionPayload,
+	Chain,
+	ChainId,
+	ChainImplementation,
+	serializeActionPayload,
+	serializeSessionPayload,
+	Session,
+	SessionPayload,
+} from "@canvas-js/interfaces"
+import { ethers } from "ethers"
+import { configure as configureStableStringify } from "safe-stable-stringify"
+import { decodeSignature, rawSecp256k1PubkeyToRawAddress, Secp256k1Wallet, serializeSignDoc } from "@cosmjs/amino"
+import { Random, Secp256k1, Secp256k1Signature, Sha256 } from "@cosmjs/crypto"
+import { toBech32 } from "@cosmjs/encoding"
+import { EvmMetaMaskSigner } from "./signerInterface.js"
+import { getActionSignatureData } from "./signatureData.js"
+
+type Secp256k1WalletPrivateKey = Uint8Array
+
+const sortedStringify = configureStableStringify({
+	bigint: false,
+	circularValue: Error,
+	strict: true,
+	deterministic: true,
+})
+
+const encodeEthAddressAsBech32 = (bech32Prefix: string, address: string) =>
+	toBech32(bech32Prefix, ethers.utils.arrayify(address))
+
+export class EvmosChainImplementation implements ChainImplementation<EvmMetaMaskSigner, Secp256k1WalletPrivateKey> {
+	public readonly chain: Chain = "cosmos"
+
+	// chainId is unused, so we can default to anything here
+	constructor(public readonly chainId: ChainId = "evmos_9001-1", public readonly bech32Prefix: string = "evmos") {}
+
+	async verifyAction(action: Action): Promise<void> {
+		if (action.session) {
+			const cosmosPrefix = "cosmos"
```

Comment:
fwiw, I think this will break tests because the bech32 address returned from `addressFromPubkey` will be specific to the given chain, and not cosmos